kafka-connect-blockchain/src/main/scala/com/datamountaineer/streamreactor/connect/blockchain/source/BlockchainSourceTask.scala

```diff
@@ -79,12 +66,7 @@ class BlockchainSourceTask extends SourceTask with StrictLogging {
   override def poll(): util.List[SourceRecord] = {
     val records = blockchainManager.map(_.get()).getOrElse(new util.ArrayList[SourceRecord]())
     logger.debug(s"Returning ${records.size()} record(-s) from Blockchain source")
-    records.foreach(r => counter.put(r.topic(), counter.getOrElse(r.topic(), 0L) + 1L))
-    val newTimestamp = System.currentTimeMillis()
-    if (counter.nonEmpty && scala.concurrent.duration.SECONDS.toSeconds(newTimestamp - timestamp) >= 60) {
-      logCounts()
-    }
-    timestamp = newTimestamp
+    progressCounter.update(records.toVector)
```

Comment:
you need to import scala.collection.JavaConversions._ for it to work without .toVector. but it will still wrap the underlying Java ArrayList to a Scala Seq,. It will use a JListWrapper in this case which will be cheaper than creating a  Scala vector